spec/store_spec.ts

```diff
@@ -0,0 +1,344 @@
+
+    devtools.dispatch(StoreDevtoolActions.toggleAction(2));
+    expect(store.getValue()).toBe(1);
+  });
+
+  it('should sweep disabled actions', () => {
+    // actionId 0 = @@INIT
+    store.dispatch({ type: 'INCREMENT' });
+    store.dispatch({ type: 'DECREMENT' });
+    store.dispatch({ type: 'INCREMENT' });
+    store.dispatch({ type: 'INCREMENT' });
+
+    expect(store.getValue()).toBe(2);
+    expect(devtools.state$.getValue().stagedActionIds).toEqual([0, 1, 2, 3, 4]);
+    expect(devtools.state$.getValue().skippedActionIds).toEqual([]);
+
+    devtools.dispatch(StoreDevtoolActions.toggleAction(2));
+    expect(store.getValue()).toBe(3);
+    expect(devtools.state$.getValue().stagedActionIds).toEqual([0, 1, 2, 3, 4]);
+    expect(devtools.state$.getValue().skippedActionIds).toEqual([2]);
+
+    devtools.dispatch(StoreDevtoolActions.sweep());
+    expect(store.getValue()).toBe(3);
+    expect(devtools.state$.getValue().stagedActionIds).toEqual([0, 1, 3, 4]);
+    expect(devtools.state$.getValue().skippedActionIds).toEqual([]);
+  });
+
+  it('should jump to state', () => {
+    store.dispatch({ type: 'INCREMENT' });
+    store.dispatch({ type: 'DECREMENT' });
+    store.dispatch({ type: 'INCREMENT' });
+    expect(store.getValue()).toBe(1);
+
+    devtools.dispatch(StoreDevtoolActions.jumpToState(0));
+    expect(store.getValue()).toBe(0);
+
+    devtools.dispatch(StoreDevtoolActions.jumpToState(1));
+    expect(store.getValue()).toBe(1);
+
+    devtools.dispatch(StoreDevtoolActions.jumpToState(2));
+    expect(store.getValue()).toBe(0);
+
+    store.dispatch({ type: 'INCREMENT' });
+    expect(store.getValue()).toBe(0);
+
+    devtools.dispatch(StoreDevtoolActions.jumpToState(4));
+    expect(store.getValue()).toBe(2);
+  });
+
+  it('should replace the reducer', () => {
+    store.dispatch({ type: 'INCREMENT' });
+    store.dispatch({ type: 'DECREMENT' });
+    store.dispatch({ type: 'INCREMENT' });
+    expect(store.getValue()).toBe(1);
+
+    store.replaceReducer(doubleCounter);
+    expect(store.getValue()).toBe(2);
+  });
+
+  it('should catch and record errors', () => {
+    spyOn(console, 'error');
+    let { store, devtools } = createStore(counterWithBug);
+
+    store.dispatch({ type: 'INCREMENT' });
+    store.dispatch({ type: 'DECREMENT' });
+    store.dispatch({ type: 'INCREMENT' });
+
+    let { computedStates } = devtools.state$.getValue();
+    expect(computedStates[2].error).toMatch(
+      /ReferenceError/
+    );
+    expect(computedStates[3].error).toMatch(
+      /Interrupted by an error up the chain/
+    );
+
+    expect(console.error).toHaveBeenCalled();
+  });
+
+  it('should catch invalid action type', () => {
+    expect(() => {
+      store.dispatch({ type: undefined });
+    }).toThrowError(
+      'Actions may not have an undefined "type" property. ' +
+      'Have you misspelled a constant?'
+    );
+  });
+
+  xit('should return the last non-undefined state from getValue', () => {
```

Comment:
I can't get this test to pass for some reason. I'm thinking there might be subtle differences between `scan` and `reduce` that I'm not taking into account.